versions/1.21.5/src/main/java/at/hannibal2/skyhanni/features/foraging/ForagingTracker.kt

```diff
@@ -236,22 +259,59 @@ object ForagingTracker : SkyHanniBucketedItemTracker<ForagingTrackerLegacy.TreeT
             }
         }
 
-        if (!openBonusGiftLoop) return
-        ForagingTrackerLegacy.bonusGiftRewardPattern.matchMatcher(message) {
-            val item = group("item")
-            val itemInternalName = ForagingTrackerLegacy.enchantedBookPattern.matchMatcher(item) {
-                val book = group("book")
-                val tier = group("tier").romanToDecimal()
-                NeuInternalName.fromItemNameOrNull("$book $tier")
-            } ?: NeuInternalName.fromItemNameOrNull(item) ?: return@matchMatcher
-
-            // Stretching Sticks handled separately due to stack size not being given in message
-            if (itemInternalName == STRETCHING_STICKS) return@matchMatcher
+        if (openBonusGiftLoop) {
+            ForagingTrackerLegacy.bonusGiftRewardPattern.matchMatcher(message) {
+                val item = group("item")
+                var itemInternalName = ForagingTrackerLegacy.enchantedBookPattern.matchMatcher(item) {
+                    val book = group("book")
+                    val tier = group("tier").romanToDecimal()
+                    NeuInternalName.fromItemNameOrNull("$book $tier")
+                } ?: NeuInternalName.fromItemNameOrNull(item) ?: return@matchMatcher
+
+                /**
+                 * this is a failsafe in the event of runes lackin' NEU repo sufficient data to automagically
+                 * fetch their correct internal names, and thus translatin' their in-game names into internal
+                 * names literally
+                 */
+                if (itemInternalName.startsWith(("◆_")))
+                    itemInternalName = itemInternalName.replace("◆_", "AXE_")
+
+                loot.addOrPut(itemInternalName, 1)
+
+                if (bonusDropsUncommonDropsList.contains(itemInternalName)
+                    && config.compactGiftBonusDropsList.get().contains(ForagingTrackerConfig.TreeGiftBonusDropCategory.UNCOMMON_DROPS)) {
+                    rareDrops.add(item)
+                }
+                if (bonusDropsEnchantedBooksList.contains(itemInternalName)
+                    && config.compactGiftBonusDropsList.get().contains(ForagingTrackerConfig.TreeGiftBonusDropCategory.ENCHANTED_BOOKS)) {
+                    rareDrops.add(item)
+                }
+                if (bonusDropsBoostersList.contains(itemInternalName)
+                    && config.compactGiftBonusDropsList.get().contains(ForagingTrackerConfig.TreeGiftBonusDropCategory.BOOSTERS)) {
+                    rareDrops.add(item)
+                }
+                if (bonusDropsShardsList.contains(itemInternalName)
```

Comment:
Not necessary for this PR if you don't want to:
Eventually, another improvement we can do to avoid code duplication is to map the category to a list of neu internal name once (in the repo reload event) then you can iterate this map once instead of the 6 if checks.

Reply:
update: okay so they're all foreach loops basically

this kinda sucks but I can work with it